pallets/uniques/src/lib.rs

```diff
@@ -0,0 +1,850 @@
+		/// - `admin`: The admin of this collection. The admin is the initial address of each
+		/// member of the collection's admin team.
+		///
+		/// Emits `Created` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::create())]
+		pub fn create(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			admin: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::create(origin, collection, admin)
+		}
+		/// Issue a new collection of non-fungible items from a privileged origin.
+		///
+		/// This new collection has no items initially.
+		///
+		/// The origin must conform to `ForceOrigin`.
+		///
+		/// Unlike `create`, no funds are reserved.
+		///
+		/// - `collection`: The identifier of the new item. This must not be currently in use.
+		/// - `owner`: The owner of this collection of items. The owner has full superuser
+		///   permissions
+		/// over this item, but may later change and configure the permissions using
+		/// `transfer_ownership` and `set_team`.
+		///
+		/// Emits `ForceCreated` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::force_create())]
+		pub fn force_create(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			owner: AccountIdLookupOf<T>,
+			free_holding: bool,
+		) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::force_create(origin, collection, owner, free_holding)
+		}
+
+		/// Destroy a collection of fungible items.
+		///
+		/// The origin must conform to `ForceOrigin` or must be `Signed` and the sender must be the
+		/// owner of the `collection`.
+		///
+		/// - `collection`: The identifier of the collection to be destroyed.
+		/// - `witness`: Information on the items minted in the collection. This must be
+		/// correct.
+		///
+		/// Emits `Destroyed` event when successful.
+		///
+		/// Weight: `O(n + m)` where:
+		/// - `n = witness.items`
+		/// - `m = witness.item_metadatas`
+		/// - `a = witness.attributes`
+		#[pallet::call_index(2)]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::destroy(
+			witness.items,
+ 			witness.item_metadatas,
+			witness.attributes,
+ 		))]
+		pub fn destroy(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			witness: DestroyWitness,
+		) -> DispatchResultWithPostInfo {
+			let collection_owner = pallet_uniques::Pallet::<T, I>::collection_owner(collection)
+				.ok_or(Error::<T, I>::UnknownCollectionOwner)?;
+			for (item, extra_deposit) in ItemExtraDeposits::<T, I>::drain_prefix(collection) {
+				let item_owner =
+					pallet_uniques::Pallet::<T, I>::owner(collection, item).ok_or(Error::<T, I>::UnknownItemOwner)?;
+				<T as pallet_uniques::Config<I>>::Currency::unreserve(&collection_owner, extra_deposit);
+				<T as pallet_uniques::Config<I>>::Currency::transfer(
```

Comment:
I think extra deposit should be owned by owner as reserved tokens, shouldn't it?

Reply:
When there is no collection, there is no reason for keeping a fund reserved. This is the unreserve condition. Otherwise we may lock tokens forever